### ISSUE TITLE
CollectionView item source is not cleared in tear down

### ISSUE THREAD

**Issue by domneedham** [AdamEssenmacher/MemoryToolkit.Maui#15]
I have a collection view that was not getting cleared correctly in tear down. 

A `CollectionView` is not of type `ListView`, so the area of code that clears a list views item source wasn't being hit. 

I think this could change to use the common underlying structure `ItemsView`, which has `ItemsSource` and `ItemTemplate`.  This then would clear down collection view and list views correctly, along with some other controls that likely inherit `ItemsView`.

I therefore propose the following:

```csharp
if (vte is ListView listView)
    listView.ItemsSource = null;
```

becomes:

```csharp
if (vte is ItemsView itemsView)
{
    itemsView.ItemsSource = null;
    itemsView.ItemTemplate = null;
}
```

This seems to sort the issue out for me.

**Comment by AdamEssenmacher** [comment]
Agreed. You can open a PR or I'll try to get to in when I'm in there next.

**Comment by vecalion** [comment]
@domneedham, the change breaks the code for the `ListView` type since the underlying structure for list views is `ItemsView<Cell>`, not `ItemsView`.
